offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/products/ProductsDbConnector.groovy

```diff
@@ -273,23 +273,59 @@ class ProductsDbConnector implements ArchiveProductDataSource, CreateProductData
    * @throws ProductExistsException if the product already exists in the data source
    */
   @Override
-  void store(Product product) throws DatabaseQueryException, ProductExistsException {
+  ProductId store(Product product) throws DatabaseQueryException, ProductExistsException {
     Connection connection = provider.connect()
 
+    ProductId id = createProductId(product)
+
     connection.withCloseable {
       PreparedStatement preparedStatement = it.prepareStatement(Queries.INSERT_PRODUCT)
       preparedStatement.setString(1, getProductType(product))
       preparedStatement.setString(2, product.description)
       preparedStatement.setString(3, product.productName)
       preparedStatement.setDouble(4, product.unitPrice)
       preparedStatement.setString(5, product.unit.value)
-      preparedStatement.setString(6, product.productId.toString())
+      preparedStatement.setString(6, id.toString())
 
       preparedStatement.execute()
     }
+
+    return id
   }
 
-/**
+  private ProductId createProductId(Product product){
+    String productType = product.productId.type
+    String version = fetchLatestVersion(product) //todo exchange with long
+
+    return new ProductId(productType,version)
+  }
+
+  private Long fetchLatestVersion(Product product){
+    String query = Queries.SELECT_ALL_PRODUCTS + "WHERE category = ?"
```

Comment:
```suggestion
    String query = Queries.SELECT_ALL_PRODUCTS + "WHERE category = ? AND <runningNumberField> GREATER ?"
```
You can leave the filtering to the database here. 
Or even directly query for the MAX value
```suggestion
    String query = "MAX(<runningNumberField>) FROM <productTable> WHERE category = ?"
```

Reply:
we do not have a runningNumberField, we store the Identifier as String. But you are right, probably you can also sort that 🤔

Reply:
What prevents us from switching to a long in the database? the parsing can be done in the db connector

Reply:
Updated the query to now do the filtering, that was a good idea